cocos/renderer/CCGLProgram.cpp

```diff
@@ -935,5 +913,20 @@ void GLProgram::reset()
     _hashForUniforms.clear();
 }
 
+void GLProgram::clearShader()
+{
+    if (_vertShader)
+    {
+        glDeleteShader(_vertShader);
+    }
+
+    if (_fragShader)
+    {
+        glDeleteShader(_fragShader);
+    }
+
+    _vertShader = _fragShader = 0;
+}
```

Comment:
I think it is better to remove them in mobile platform, because it will consume some memory.